ProjectName/src/screens/PaymentDetails/PaymentDetailsForm/components/customAmountInput/CustomAmountInput.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {FC} from 'react';
+import {Controller} from 'react-hook-form';
+import {View, Text, TextInput, Image} from 'react-native';
+
+import {DECIMAL_REGEX} from './custom-amount-input.constants';
+import {styles} from './custom-amount-input.styles';
+import {CustomMessageInputTypes} from '../customMessageInput/custom-message-input.types';
+
+const CustomAmountInput: FC<CustomMessageInputTypes> = ({
+  control,
+  name,
+  title,
+  placeholder,
+}) => {
+  return (
+    <View>
+      <Text style={styles.title}>{title}</Text>
+      <Controller
+        control={control}
+        name={name}
+        render={({field: {value, onChange, onBlur}}) => (
+          <View style={styles.container}>
+            <TextInput
+              keyboardType="decimal-pad"
+              value={value}
+              onChangeText={text => {
+                if (DECIMAL_REGEX.test(text)) {
+                  const formattedValue = text;
```

Comment:
Just curios, what's is the purpose of creating a separate variable here. Maybe we don't need it?